diff --git a/libvlc/compile.sh b/libvlc/compile.sh
index e3a2a1133..c2571a67e 100755
--- a/libvlc/compile.sh
+++ b/libvlc/compile.sh
@@ -2,11 +2,14 @@
 
 set -e
 
+# TODO: pdb mapped to source path
 usage()
 {
-    echo "Usage: compile <arch> <TargetOS>"
+    echo "Usage: compile <arch> <TargetOS> <Release/Debug> <PDBs>"
     echo "archs: i686,x86_64,armv7,aarch64"
     echo "os: win10"
+    echo "Release/Debug: release,debug"
+    echo "PDBs: yes,no"
 }
 
 using()
@@ -64,11 +67,39 @@ case "$2" in
         ;;
 esac
 
+case "$3" in
+    release)
+        DEBUG=OFF
+        ;;
+    debug)
+        DEBUG=ON
+        ;;
+    *)
+        echo "Unknown Release/Debug option"
+        usage
+        exit 1
+        ;;
+esac
+
+case "$4" in
+    yes)
+        PDB=ON
+        ;;
+    no)
+        PDB=OFF
+        ;;
+    *)
+        echo "Unknown PDB option"
+        usage
+        exit 1
+        ;;
+esac
+
 # 1/ libvlc, libvlccore and its plugins
 TESTED_HASH=45df8a6415
 if [ ! -d "vlc" ]; then
     echo "VLC source not found, cloning"
-    git clone http://git.videolan.org/git/vlc/vlc-3.0.git vlc
+    git clone -b 3.0.x https://code.videolan.org/videolan/vlc.git vlc
     cd vlc
     git config --global user.email "cone@example.com"
     git config --local user.name "Cony Cone"
@@ -144,10 +175,26 @@ else
 fi
 
 echo "Building the contribs"
+
+VLC_ROOT_PATH="$( cd "$(dirname "$0")" ; pwd -P )"
+
+CONTRIB_BOOTSTRAP_OPTIONS=
+if [ "${PDB}" = "ON" ]; then
+    CONTRIB_BOOTSTRAP_OPTIONS="${CONTRIB_BOOTSTRAP_OPTIONS} --enable-pdb"
+    PDB_MAP="c:/vlc"
+    CFLAGS="$CFLAGS -fdebug-prefix-map='$VLC_ROOT_PATH'='$PDB_MAP'"
+    CXXFLAGS="$CXXFLAGS -fdebug-prefix-map='$VLC_ROOT_PATH'='$PDB_MAP'"
+    export CFLAGS
+    export CXXFLAGS
+fi
+if [ "${DEBUG}" = "ON" ]; then
+    CONTRIB_BOOTSTRAP_OPTIONS="${CONTRIB_BOOTSTRAP_OPTIONS} --disable-optim"
+fi
+
 CONTRIB_FOLDER=contrib/winrt-$1-$RUNTIME
 mkdir -p $CONTRIB_FOLDER
 cd $CONTRIB_FOLDER
-../bootstrap --host=${TARGET_TUPLE} --build=$BUILD_ARCH --disable-disc \
+../bootstrap --host=${TARGET_TUPLE} --build=$BUILD_ARCH ${CONTRIB_BOOTSTRAP_OPTIONS} --disable-disc \
     --disable-sdl \
     --disable-schroedinger \
     --disable-vncserver \
@@ -202,20 +249,33 @@ echo "Bootstraping"
 ../bootstrap
 
 echo "Configuring"
+
+CONFIGURE_OPTIONS=
+if [ "${PDB}" = "ON" ]; then
+    CONFIGURE_OPTIONS="${CONFIGURE_OPTIONS} --enable-pdb"
+fi
+if [ "${DEBUG}" = "ON" ]; then
+    CONFIGURE_OPTIONS="${CONFIGURE_OPTIONS} --enable-debug"
+fi
+
 CPPFLAGS="${EXTRA_CPPFLAGS}" \
 LDFLAGS="${EXTRA_LDFLAGS}" \
 CC="${COMPILER}" \
 CXX="${COMPILERXX}" \
 ac_cv_search_connect="-lws2_32" \
-../../configure.sh --host=${TARGET_TUPLE}
+../../configure.sh --host=${TARGET_TUPLE} ${CONFIGURE_OPTIONS}
 
 echo "Building"
 make $MAKEFLAGS
 
 echo "Package"
-make install
+if [ "${PDB}" = "ON" ]; then
+    make package-win-install
+else
+    make install
+fi
 
-rm -rf tmp && mkdir tmp
+rm -rf tmp && mkdir -p tmp/plugins
 
 # Compiler shared DLLs, when using compilers built with --enable-shared
 # The shared DLLs may not necessarily be in the first LIBRARY_PATH, we
@@ -223,8 +283,13 @@ rm -rf tmp && mkdir tmp
 library_path_list=`${TARGET_TUPLE}-g++ -v /dev/null 2>&1 | grep ^LIBRARY_PATH|cut -d= -f2` ;
 
 find _win32/bin -name "*.dll" -exec cp -v {} tmp \;
-cp -r _win32/include tmp/
 cp -r _win32/lib/vlc/plugins tmp/
+cp -r _win32/include tmp/
+if [ "${PDB}" = "ON" ]; then
+    find _win32/bin -name "*.pdb" -exec cp -v {} tmp \;
+else
+    find tmp/plugins -name "*.pdb" -exec rm -v {} \;
+fi
 
 find tmp -name "*.la" -exec rm -v {} \;
 find tmp -name "*.a" -exec rm -v {} \;
@@ -263,6 +328,9 @@ do
     fi
 done
 rm `find tmp/plugins -name 'lib*plugin.dll' | grep -E "lib(${regexp})_plugin.dll"`
+if [ "${PDB}" = "ON" ]; then
+    rm `find tmp/plugins -name 'lib*plugin.pdb' | grep -E "lib(${regexp})_plugin.pdb"`
+fi
 
 find tmp \( -name "*.dll" -o -name "*.exe" \) -exec ${TARGET_TUPLE}-strip {} \;
 find tmp \( -name "*.dll" -o -name "*.exe" \) -exec ../../appcontainer.pl {} \;
@@ -271,9 +339,18 @@ cp lib/.libs/libvlc.dll.a tmp/libvlc.lib
 cp src/.libs/libvlccore.dll.a tmp/libvlccore.lib
 
 CURRENTDATE="$(date +%Y%m%d-%H%M)"
+ARCHIVE_NAME=vlc-${1}-${2}-${CURRENTDATE}
+if [ "${DEBUG}" = "ON" ]; then
+    ARCHIVE_NAME="${ARCHIVE_NAME}-debug"
+else
+    ARCHIVE_NAME="${ARCHIVE_NAME}-release"
+fi
+if [ "${PDB}" = "ON" ]; then
+    ARCHIVE_NAME="${ARCHIVE_NAME}-pdb"
+fi
 
 cd tmp
-7z a ../vlc-${1}-${2}-${CURRENTDATE}.7z *
+7z a ../${ARCHIVE_NAME}.7z *
 cd ..
 rm -rf tmp
 
